src/factories/nodes.ts

```diff
@@ -53,6 +53,14 @@ export async function nodesContainerFactory() {
     }
   })
 
+  emitter.on('nodeSelected', (selection) => {
+    unHighlightPath()
+
+    if (selection?.id && nodes.has(selection.id)) {
+      highlightPath(selection.id)
+    }
+  })
```

Comment:
This is looping every node and every edge twice but we could do it with a single loop. Rather than unHighlighting lets loop and set the opacity based on if the edge should be highlighted or not. If nothing should be highlighted the pathIds can just be an empty array